src/account/types/mod.rs

```diff
@@ -112,15 +112,18 @@ impl OutputData {
     ) -> crate::Result<InputSigningData> {
         let unlock_address = {
             if let Some(unlock_conditions) = self.output.unlock_conditions() {
-                *unlock_conditions.locked_address(&self.address, current_time)
+                match &self.output {
+                    Output::Foundry(foundry) => Address::Alias(*foundry.alias_address()),
```

Comment:
What is in `self.address` in case of a foundry?

Reply:
The ed25519 address for which we got the alias output, which then owns the foundry output